pokemon-app/src/components/SinglePokemon.js

```diff
@@ -118,36 +83,12 @@ const SinglePokemon = ({
                 Your Pokemon
               </div>
 
-              <div
-                className={`moves-box ${hasEnemy ? "has-enemy" : ""} ${
-                  statsOnTop ? "" : "on-top"
-                }`}
-                ref={scrollRef}
-                onScroll={onScroll}
-              >
-                <ol id="moves-list">
-                  {!pokemon?.moves
-                    ? null
-                    : pokemon?.moves.map((moveData, moveIndex) => (
-                        <li key={moveIndex}>{moveData.move.name}</li>
-                      ))}
-                </ol>
-              </div>
-
-              {renderMovesListTopArrow() && (
-                <span
-                  className={`moves-list--up-arrow ${
-                    hasEnemy ? "has-enemy" : ""
-                  }`}
-                ></span>
-              )}
-
-              {renderMovesListBottomArrow() && (
-                <span
-                  className={`moves-list--down-arrow ${
-                    hasEnemy ? "has-enemy" : ""
-                  }`}
-                ></span>
+              {checkDoesMovesKeyExistInObject() && (
+                <MovesList
+                  pokemonMoves={pokemon?.moves}
+                  hasEnemy={hasEnemy}
+                  statsOnTop={statsOnTop}
```

Comment:
I see you moved the logic into a component! Looks great. I am sure you already had this idea, without me saying anything. Get functionality first, then refactor right?

Reply:
@krferrell Yas. Make the machine work before doing refactors :)